### DISCUSSION TITLE
Create an admin panel

### DISCUSSION THREAD

**Discussion by BenNotix** [Just-Moh-it/Pckd#43] [Version 2]
Hello! 👋
First of all, thank you for offering this service for free!

I think it would be interesting to create an administration panel to manage all the URLs present on our server of the different users. 
Moreover, it would be also interesting to make sure that anonymous users can't create links (for example, put an option in the admin panel to allow or not this). 
Also, a system to manage the users present on our instance, to limit it to certain users only. For example, an approval system for users when they register on the panel could be a good idea. 

If you have any questions about these ideas, please do not hesitate to contact me. 
Thanks again to you, 💪
Ben.

**Comment by ItsNoted** [comment]
I just installed this and also cannot figure out how to log into the admin panel. There are no clear instructions on this. Also I cant edit, trash or modify any of the links from the user dashboard either.

I agree, we need some control over users and moderation of links being added via admin panel some how.

**Comment by Just-Moh-it** [comment]
👋 Hey, thanks for taking out the time to check out Pckd!

Creating an admin panel, restricting new-user access, limiting anonymous access etc. were always on the to-do list for the app from the beginning... However, due to this project not gaining enough attraction/users, I kept it for later so that I could invest time on other projects.

Looks like the project is now getting the required attention and has just got its first sponsor (thanks @BenNotix, I'll add you to the README shortly). 

For now, I will focus on adding edit and delete buttons to users accounts, adding a setting to restrict access for anonymous users, adding better filtering/sorting support, optimizing the API calls, adding web sockets, and adding a setup screen.

After a while, I'll add the user-role system and email verification to the app. I'm planning to release a v-2 of this created using NextJS, but that seems far in the future. For now, the current implementation would do.

Again, thanks for trying out Pckd! 😃 I'll get back once the features are implemented.

**Comment by ItsNoted** [comment]
> 👋 Hey, thanks for taking out the time to check out Pckd!
> 
> Creating an admin panel, restricting new-user access, limiting anonymous access etc. were always on the to-do list for the app from the beginning... However, due to this project not gaining enough attraction/users, I kept it for later so that I could invest time on other projects.
> 
> Looks like the project is now getting the required attention and has just got its first sponsor (thanks @BenNotix, I'll add you to the README shortly).
> 
> For now, I will focus on adding edit and delete buttons to users accounts, adding a setting to restrict access for anonymous users, adding better filtering/sorting support, optimizing the API calls, adding web sockets, and adding a setup screen.
> 
> After a while, I'll add the user-role system and email verification to the app. I'm planning to release a v-2 of this created using NextJS, but that seems far in the future. For now, the current implementation would do.
> 
> Again, thanks for trying out Pckd! 😃 I'll get back once the features are implemented.

Excellent. Sounds like you’re keeping busy with projects! Thanks for the update and for sharing your plans for the next release. Also, you mentioned this project not gaining enough attraction. It has more stars than any of your other projects on your profile! ;)

**Comment by rizwan95** [comment]
Come on @Just-Moh-it , we need an admin panel :D

**Comment by Doncellup** [comment]
> Hello! 👋 First of all, thank you for offering this service for free!
> 
> I think it would be interesting to create an administration panel to manage all the URLs present on our server of the different users. Moreover, it would be also interesting to make sure that anonymous users can't create links (for example, put an option in the admin panel to allow or not this). Also, a system to manage the users present on our instance, to limit it to certain users only. For example, an approval system for users when they register on the panel could be a good idea.
> 
> If you have any questions about these ideas, please do not hesitate to contact me. Thanks again to you, 💪 Ben.

All the suggestions he made are awesome!! I look forward to them!
